matrix_client/client.py

```diff
@@ -251,32 +266,31 @@ def _mkroom(self, room_id):
     def _process_state_event(self, state_event, current_room):
         if "type" not in state_event:
             return  # Ignore event
-
         etype = state_event["type"]
 
         if etype == "m.room.name":
-            current_room.name = state_event["content"]["name"]
+            current_room.name = state_event["content"].get("name", None)
         elif etype == "m.room.topic":
-            current_room.topic = state_event["content"]["topic"]
+            current_room.topic = state_event["content"].get("topic", None)
         elif etype == "m.room.aliases":
-            current_room.aliases = state_event["content"]["aliases"]
+            current_room.aliases = state_event["content"].get("aliases", None)
 
-    def _sync(self, limit=1):
-        response = self.api.initial_sync(limit)
-        try:
-            self.end = response["end"]
-            for room in response["rooms"]:
-                self._mkroom(room["room_id"])
+    def _sync(self, timeout_ms=30000):
+        # TODO: Deal with presence
+        # TODO: Deal with left rooms
+        response = self.api.sync(self.sync_token, timeout_ms)
+        self.sync_token = response["next_batch"]
+        for room_id in response['rooms']['join']:
```

Comment:
I'd prefer `response.get('rooms', {}).get('join', {})` for paranoia.

Also, why not `for room_id, room in _.items():`?